### PR TITLE
Shiwani ProjectReport unit test

### PR DESCRIPTION
**Pull request by Shiwani99**
# Description
Unit test for ProjectReport component

## Main changes explained:
- Added 10 test cases: renders without crashing, check if project name is displaying, check if getProjectDetail works as expected, check if getProjectDetail puts out an error message when get request fails, check if fetchAllMembers works expected, check if fetchAllMembers puts out an error message when get request fails, check if getProjectActiveUser works as expected, check if getProjectActiveUser puts an error message when get request fails, check if fetchAllWBS works as expected, check if fetchAllWBS puts out an error message when get request fails

## How to test:
1. check into current branch
2. do `npm test src/components/Reports/ProjectReport/_tests__/ProjectReport.test-js` 
3. check if all the test cases pass without any errors

## Screenshots or videos of changes:
ProjectReport_Unit_Test_Output

### CODE REVIEW

**Review by meetpadhiar4 — APPROVED** [review]
Hi, I checked your PR and it passed all the test cases. All the components are being rendered without any errors as well. Please refer the screenshot attached.

PR 1978

**Review by ChengyanOo — APPROVED** [review]
Everything works as expected, but the test script path should be 'src/components/Reports/ProjectReport/__tests__/ProjectReport.test.js'
Screenshot 2024-02-22 at 18 19 11

**Review by vikrambadhan — APPROVED** [review]
Hi, I tested your PR and I all the test cases passed!

Week 26 - PR4

**Review by XiaohanMeng — APPROVED** [review]
Hi, 11 test cases passed. Good job!
#1978-result

**Review by ShengjieMao — APPROVED** [review]
All tests passed. The file directed to the file in the commend might be __tests__\/ProjectReport.test.j rather than what described in the instruction. Other Components and props are well checked in the file.

**Review by skaush21 — APPROVED** [review]
unit test works, however I ran into issue testing the given npm test.
instead i used npm run test ProjectReport.test.js
all tests pass.

File: src/components/Reports/ProjectReport/__tests__/ProjectReport.test.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
```
Comment (FyisFe) [inline src/components/Reports/ProjectReport/__tests__/ProjectReport.test.js:2]: `fireEvent, waitFor` is declared but not used

File: src/components/Reports/ProjectReport/__tests__/ProjectReport.test.js [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+import '@testing-library/jest-dom/extend-expect';
+import thunk from 'redux-thunk';
+import configureStore from 'redux-mock-store';
+import { Provider } from 'react-redux';
+import ProjectReport from '..';
+import axios from 'axios';
+import { getProjectDetail } from 'actions/project';
+import { fetchAllMembers, foundUsers, getProjectActiveUser } from 'actions/projectMembers';
```
Comment (FyisFe) [inline src/components/Reports/ProjectReport/__tests__/ProjectReport.test.js:10]: `foundUsers` is declared but not used

**Review by FyisFe — APPROVED** [review]
All tests passed. Everything looks great except for some unused functions imported

image

**Comment by beblicarl** [comment]
All test cases passed successfully and all test cases are great

image

**Review by beblicarl — APPROVED** [review]

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**